engine/classes/Elgg/AjaxApi/ApiResponse.php

```diff
@@ -0,0 +1,61 @@
+<?php
+namespace Elgg\AjaxApi;
+
+/**
+ * JSON endpoint response
```

Comment:
I think we can consider making this public and allowing devs to return it from page and action handlers. That would be cleaner than echoing manually JSON-encoded values and also allow setting TTL. If pages/actions return arrays we could assume this to be data, but it doesn't seem clean to me to always require an array/object return value, but I could be convinced.